### ISSUE TITLE
Infinite scroll destroy doesn't work

### ISSUE THREAD

**Issue by Macro-Jackson** [metafizzy/infinite-scroll#871]
I've been looking at my code for 6 hours to get my crappy JS working. I'm a novice JS programmer.... 

What I do:
when the page is loaded I execute:

```
var infScroll = new InfiniteScroll('#infiniteContainer', { "path": ".next", "append": ".product", "hideNav": ".pagination", "scrollThreshold": "1000" });
```
Then when I filter my results then I do this:

```
//	Destroy InfiniteScroll instance
if (typeof infScroll !== 'undefined') {
	infScroll.destroy();
}

```
It does exist so it executes the destroy. After this it is not gone. I can still watch infScroll. 
Then I do my fetch to reload my page content and I update my infinite scroll button's href:
```
document.getElementById('searchPageNext').setAttribute('href', serverResponse.search.page.next);
```

```
//	Re-initialize InfiniteScroll instance
var infScroll = new InfiniteScroll('#infiniteContainer', { "path": ".next", "append": ".product", "hideNav": ".pagination", "scrollThreshold": "1000" });
```

What happens after this is that there are 2 infinite scroll scripts running. One on the old URL loading on each mouse tick and one working normal when you scroll down.

How can it be that when I destroy the infScroll instance that it still works after the destroy? Why can there be two or even more scripts running?

Is it because I create a new InfiniteScroll in my fetch function?

In other words, what did I do wrong?

**Comment by Macro-Jackson** [comment]
Found my stupid mistake. Local variables do not pass into anonymous functions (fetch function).... So I've made my infScroll global by making it window.infScroll.

Not sure if this is the perfect solution but it seems to work.
